textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/join/Join.java

```diff
@@ -256,24 +273,50 @@ private ITuple joinTuples(ITuple outerTuple, ITuple innerTuple, JoinPredicate jo
 			return null;
 		}
 
-		// TODO Discuss and implement as to what Schema has to be.
-		// It shouldn't just be inner tuple or outer tuple schema.
-		Schema schema = innerTuple.getSchema();
-		List<IField> fieldList = innerTuple.getFields();
-		IField[] nextTupleField = new IField[fieldList.size()];
+		List<IField> newFieldList = new ArrayList<>();
+
+		ArrayList<Attribute> innerAttrList = new ArrayList<>();
```

Comment:
Add comments to the codebase to explain the main logic here, i.e., using the intersection of the two sets of attributes as the schema of the output.